src/plugins/data_view_management/public/components/edit_index_pattern/edit_index_pattern.tsx

```diff
@@ -61,11 +62,17 @@ const securityDataView = i18n.translate(
 
 const securitySolution = 'security-solution';
 
+const getCompositeRuntimeFields = (dataView: DataView) =>
+  pickBy(dataView.getAllRuntimeFields(), (fld) => fld.type === 'composite');
+
 export const EditIndexPattern = withRouter(
   ({ indexPattern, history, location }: EditIndexPatternProps) => {
     const { uiSettings, overlays, chrome, dataViews, IndexPatternEditor, savedObjectsManagement } =
       useKibana<IndexPatternManagmentContext>().services;
     const [fields, setFields] = useState<DataViewField[]>(indexPattern.getNonScriptedFields());
+    const [compositeRuntimeFields, setCompositeRuntimeFields] = useState<
+      Record<string, RuntimeField>
+    >(getCompositeRuntimeFields(indexPattern));
```

Comment:
let's use the initial state factory here instead of a value to avoid calling `getCompositeRuntimeFields` on each render
https://reactjs.org/docs/hooks-reference.html#lazy-initial-state